Problem2/src/sorting.c

```diff
@@ -0,0 +1,78 @@
+/*
+ *
+ * sorting.c
+ *
+ */
+
+#include <stdio.h>
+#include "xparameters.h"
+#include "xgpio.h"
+#include "xil_printf.h"
+
+#define SIZE 20 
+#define SW_DEVICE_ID  XPAR_GPIO_0_DEVICE_ID
+
+XGpio SW_Gpio;
+
+void Sorting(int* data,int mode)
+{
+	/*
+	 *mode 1 for DESCENDING
+	 *mode 0 for INCREMENT
+	 */
+	int key;
+	int j;
+	for(int i=1; i<SIZE; i++)
+	{
+		key = data[i];
+		for(j = i-1; j>=0; j--)
+		{
+			if(!((key > data[j]) ^ mode))
+				data[j+1] = data[j];
+			else
+				break;
+		}
+		data[j+1] = key;
+	}
+}
+int main()
+{
+	int SW_Status;
+	int data[SIZE];
+	u32 sw_data;
+	/* Initialize the GPIO driver */
+	SW_Status = XGpio_Initialize(&SW_Gpio, SW_DEVICE_ID);
+	if (SW_Status != XST_SUCCESS)
+	{
+		xil_printf("Gpio Initialization Failed\r\n");
+		return XST_FAILURE;
+	}
+	/* Set the direction for all signals as inputs except the LED output */
+	XGpio_SetDataDirection(&SW_Gpio, 1, 0x0f);
+
+	while(1)
+	{
+		/* Scan input data*/
+		for(int i=0; i<SIZE; i++)
+		{
+			printf("Enter the No.%d Data:", i);
+			scanf("%d", &data[i]);
+			printf(" %d\r\n", data[i]);
```

Comment:
Do you think it is necessary to add some error-detection machenism, such as non-numeric input? Or maybe it's just fine for the homework? XD

BTW, maybe it doesn't have to print the user input immediately after `scanf`? Since what user has typed were still be able to be seen on the PuTTY terminal? (I'm not sure about that)